indico/web/client/js/react/components/principals/PrincipalListField.jsx

```diff
@@ -182,10 +178,33 @@ PrincipalListField.defaultProps = {
   withExternalUsers: false,
   withEventRoles: false,
   withCategoryRoles: false,
+  withRegistrants: false,
   eventId: null,
   readOnly: false,
 };
 
+export function AddPrincipalDropdown({text, options, onChange}) {
+  return (
+    <Dropdown
+      text={text}
+      button
+      upward
+      floating
+      disabled={options.length === 0}
+      options={options}
+      openOnFocus={false}
+      selectOnBlur={false}
+      selectOnNavigation={false}
+      onChange={(e, data) => onChange([{identifier: data.value}])}
+    />
+  );
+}
+AddPrincipalDropdown.propTypes = {
+  text: PropTypes.string.isRequired,
+  options: PropTypes.array.isRequired,
+  onChange: PropTypes.func.isRequired,
+};
+
 export default React.memo(PrincipalListField);
```

Comment:
Having this after `AddPrincipalDropdown` is a bit strange since it belongs to `PrincipalListField`..